src/modules/plus/filter_strobe.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * filter_strobe.c -- simple strobing filter
+ * Copyright (C) 2007 Martin Sandsmark <martin.sandsmark@kde.org>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <framework/mlt_filter.h>
+#include <framework/mlt_frame.h>
+#include <framework/mlt_producer.h>
+#include <framework/mlt_service.h>
+#include <framework/mlt_factory.h>
+#include <framework/mlt_property.h>
+
+#include <stdio.h>
+#include <string.h>
+
+static int filter_get_image( mlt_frame frame, uint8_t **image, mlt_image_format *format, int *width, int *height, int writable )
+{
+	// Get the image
+	mlt_filter filter = mlt_frame_pop_service( frame );
+	mlt_properties properties = MLT_FILTER_PROPERTIES( filter );
+
+	int error = mlt_frame_get_image( frame, image, format, width, height, 1 );
+	if (error) {
+		return error;
+	}
+
+	int invert = mlt_properties_get_int( properties, "strobe_invert" );
+	int interval = mlt_properties_get_int( properties, "interval" );
+	mlt_position currentpos = mlt_filter_get_position( filter, frame );
+
+	int do_strobe = (currentpos % (interval + 1)) > interval/2;
+	if (invert) {
+		do_strobe = !do_strobe;
+	}
+
+	if (do_strobe == 1) {
+		int alphasize = *width * *height;
+		uint8_t *alpha_buffer = mlt_frame_get_alpha_mask( frame );
```

Comment:
If the `format` is `mlt_image_rgb24a` then you need to clear that alpha channel as well because if a downstream component requests rgb24a and your service returns there is no conversion that will happen in `mlt_frame->convert_image()`.

Reply:
this might explain some of the issues I had while testing.

Reply:
Is there a preferred method of doing this, or should I just do it the old school way?

Reply:
just did it the oldschool way